src/codegen/cfg.rs

```diff
@@ -1254,6 +1257,10 @@ pub fn optimize_and_check_cfg(
     if opt.dead_storage {
         dead_storage::dead_storage(cfg, ns);
     }
+
+    if opt.common_subexpression_elimination && func_no.is_some() {
```

Comment:
Any reason for `func_no.some()`? 

A default constructor has no function number, so it is `None` in that case.

Reply:
A default constructor has no sub expression to be optimized.

Reply:
True, although that is worth a comment.

Reply:
Done